src/api/app/controllers/issue_trackers_controller.rb

```diff
@@ -1,6 +1,5 @@
 class IssueTrackersController < ApplicationController
-  before_action :require_admin, only: [:create, :update, :destroy]
-
+  prepend_before_action :require_admin, only: [:create, :update, :destroy]
```

Comment:
This won't work because it's prepending `require_admin` before the `extract_user` callback set in `ApplicationController`.

https://github.com/openSUSE/open-build-service/blob/ce8e211ee0eb52c2aba2fd556dda09471e8d7c07/src/api/app/controllers/application_controller.rb#L27

 `extract_user` is setting the user by calling `check_extracted_user`.

https://github.com/openSUSE/open-build-service/blob/ce8e211ee0eb52c2aba2fd556dda09471e8d7c07/src/api/app/lib/authenticator.rb#L198-L223

Reply:
Give it a try locally with this `osc` command:
```
osc --apiurl=http://localhost:3000 api -X POST /issue_trackers -d "<issue-tracker><name>test</name>description>My test issue tracker</description><regex>test#\d+test</regex><label>test#\@\@\@+test</label><kind>bugzilla</kind><enable-fetch>false</enable-fetch><user>obsbugbot</user><password>secret</password><url>http://example.com</url><show-url>http://example.com/@@@</show-url></issue-tracker>"
```

It will result in `Server returned an error: HTTP Error 403: Forbidden Requires admin privileges` even if you are logged-in as an admin.

Reply:
yes i had a `skip_before_action :extract_user` which was removed before pushing. Now it is fixed.